### ISSUE TITLE
Build error: quippy tries to compile using gcc in icc arch

### ISSUE THREAD

**Issue by kbsezginel** [libAtoms/QUIP#90]
Hi,

I am trying to compile Python bindings for QUIP using the `linux_x86_64_ifort_icc_openmp` arcihtecture. I was able to compile QUIP with GAP using the `-mkl` flag (which I added to the arch Makefile) for BLAS and LAPACK however when I run `make quippy` I get the following error:
```
gcc -pthread -shared -Wl,-z,relro -g build/linux_x86_64_ifort_icc_openmp/temp.linux-x86_64-2.7/arraydatamodule.o build/linux_x86_64_ifort_icc_openmp/temp.linux-x86_64-2.7/ihome/cwilmer/kbs37/venv/soap/lib/python2.7/site-packages/numpy/f2py/src/fortranobject.o -L/usr/lib64 -lpython2.7 -o build/linux_x86_64_ifort_icc_openmp/lib.linux-x86_64-2.7/quippy/arraydata.so -mkl
gcc: error: unrecognized command line option ‘-mkl’
```
I am not sure why it's using `gcc` in the first place. I have looked at the arch Makefiles however all the cpp compilers are set to `icc`. Is there another way to make it use `icc` instead of `gcc`?

Thank you,
Kutay

**Comment by gabor1** [comment]
this is just a guess (James Kermode knows quippy infinitely better than me), but it could be that quippy needs to be compiled with the same C compiler that is tied into your python environment. If your python uses gcc, then you have to use gcc.  

-- Gábor

**Comment by jameskermode** [comment]
Yes, it's the C compiler that is being chosen by Python's distutils in `setup.py` that is the problem - it is detecting gcc and not icc. Try going into the `quippy/` subdirectory and running`python setup.py build --help-compiler` to see if intel is available, then select with e.g. `python setup.py build --compiler=intel`. There's a similar argument for Fortran compiler (`--fcompiler`, I think).

**Comment by jameskermode** [comment]
(We should probably pass along the settings from arch Makefiles, but the names don't match up so it's a bit tricky)

**Comment by kbsezginel** [comment]
Thanks, it seems like the C compiler works fine however, now I get the following error with `ifort`.
I selected intel compilers for 64-bit for both C and Fortran as: `python setup.py install --compiler=intelem --fcompiler=intelem`.

```
ld: cannot find -lquiputils
ld: cannot find -lquip_core
ld: cannot find -lgap_filler
ld: cannot find -lgap_predict
ld: cannot find -latoms
ld: cannot find -lFoX_sax
ld: cannot find -lFoX_wxml
ld: cannot find -lFoX_utils
ld: cannot find -lFoX_common
ld: cannot find -lFoX_fsys
ld: cannot find -lquiputils
ld: cannot find -lquip_core
ld: cannot find -lgap_filler
ld: cannot find -lgap_predict
ld: cannot find -latoms
ld: cannot find -lFoX_sax
ld: cannot find -lFoX_wxml
ld: cannot find -lFoX_utils
ld: cannot find -lFoX_common
ld: cannot find -lFoX_fsys
```
Do I need to compile `numpy` with intel compilers as well?

**Comment by jameskermode** [comment]
No, this looks like a path problem finding the QUIP libraries. If you change how `setup.py` is invoked from `${QUIP_ROOT}/quippy/Makefile` to include your new flags then `make quippy` from `${QUIP_ROOT}` should work.

**Comment by kbsezginel** [comment]
Thanks, I was able to compile and install `quippy` this way, however, when I run the tests now with `make test` I get the following error:
```
test_force (test_atoms.TestAtoms_Rotate) ... Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
```
I believe this is `numpy` having trouble finding the `mkl` libraries. I tried sourcing the library, reinstalling `numpy`, creating a fresh virtual environment etc. but with not luck. Any idea how to resolve this?

I think at this point I will try using `gcc` and figure out the correct `mkl` flags for my system.
